### DISCUSSION TITLE
Extending auto generated entity files

### DISCUSSION THREAD

**Discussion by Qwertie-** [SeaQL/sea-orm#1309] [Q&A]
We are using `sea-orm-cli generate entity` to generate entity files. Of course this means they can no longer be manually modified. I'm wondering if there are any established patters or good ideas for how you could have custom functions added to entities. Imagine a `full_name` function for example where you could call `user.full_name()` on a user `Model` and have it return first_name + last_name.

Any ideas or thinking around this?

**Comment by billy1624** [comment]
Hey @Qwertie-, welcome to SeaQL community.

You can implement the `full_name` method for the `Model`

```rs
impl Model {
    pub fn full_name(&self) -> String {
        format!("{} {}", self.first_name, self.last_name)
    }
}
```

**Reply by billy1624** [reply]
Oh wait. Let say you output the generated entity files into the `/entity/*` folder. Then, you can create another module under the same crate and implement the above.

This way, you leave the generated entity files untouched while adding customized code on top of it.

**Reply by AngelOnFira** [reply]
This is pretty cool, I've always looked at the entity crate as "generated only". I haven't thought to put helpers somewhere else in that crate.
